hpOneView/image_streamer/resources/artifact_bundles.py

```diff
@@ -0,0 +1,375 @@
+
+        Args:
+            start:
+                The first item to return, using 0-based indexing.
+                If not specified, the default is 0 - start with the first available item.
+            count:
+                The number of resources to return. A count of -1 requests all items.
+
+                The actual number of items in the response might differ from the requested
+                count if the sum of start and count exceeds the total number of items.
+            filter (list or str):
+                A general filter/query string to narrow the list of items returned. The
+                default is no filter; all resources are returned.
+            sort:
+                The sort order of the returned data set. By default, the sort order is based
+                on create time with the oldest entry first.
+
+        Returns:
+            list: A list of Artifacts Bundle.
+        """
+        return self._client.get_all(start, count, filter=filter, sort=sort)
+
+    def get(self, id_or_uri):
+        """
+        Retrieves the overview details of the selected Artifact Bundle as per the selected attributes.
+
+        Args:
+            id_or_uri: ID or URI of the OS Volume.
+
+        Returns:
+            dict: The Artifact Bundle.
+        """
+        return self._client.get(id_or_uri)
+
+    def get_by(self, field, value):
+        """
+        Gets all Artifacts Bundle that match the filter.
+
+        The search is case-insensitive.
+
+        Args:
+            field: Field name to filter.
+            value: Value to filter.
+
+        Returns:
+            dict: The Artifacts Bundle.
+        """
+        return self._client.get_by(field, value)
+
+    def get_by_name(self, name):
+        """
+        Gets an Artifact Bundle by name.
+
+        Args:
+            name: Name of the Artifact Bundle.
+
+        Returns:
+            dict: The Artifact Bundle.
+        """
+        return self._client.get_by_name(name)
+
+    def get_backups(self):
+        """
+        Get all Backups for Artifact Bundle.
+
+        Returns:
+            list: A list of Backups for Artifacts Bundle.
+        """
+        return self._client.get(id_or_uri=self.URI + self.BACKUPS_PATH)
+
+    def get_backups_by_id(self, id_or_uri):
+        """
+        Get the details of the backup for an Artifact Bundle.
+
+        Args:
+            id_or_uri: ID or URI of the Artifact Bundle.
+
+        Returns:
+            list: Backup for an Artifacts Bundle.
+        """
+        uri = self.URI + self.BACKUPS_PATH + '/' + extract_id_from_uri(id_or_uri)
+        return self._client.get(id_or_uri=uri)
+
+    def download_archive_artifact_bundle(self, id_or_uri, file_path):
+        """
+        Download archive for the Artifact Bundle.
+
+        Args:
+            id_or_uri: ID or URI of the Artifact Bundle.
+            file_path(str): Destination file path.
+
+        Returns:
+            list: A list of Artifacts Bundle.
```

Comment:
Looks like this method should return something different than a list of Artifacts Bundle